src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -0,0 +1,79 @@
+import { Position, Range, TextEditor } from "vscode";
+import { getMatcher } from "../../../core/tokenizer";
+import { Target } from "../../../typings/target.types";
+import { getTokensInRange } from "../../../util/getTokensInRange";
+import { expandToFullLine } from "../../../util/rangeUtils";
+import { TokenTarget } from "../../targets";
+import { ContainingIndices, ScopeHandler } from "./scopeHandler.types";
+
+export default class TokenScopeHandler extends ScopeHandler {
+  protected getEveryTarget(
+    editor: TextEditor,
+    contentRange: Range,
+    isReversed: boolean,
+    hasExplicitRange: boolean
+  ): Target[] {
+    const tokenRanges = getTokensInRange(
+      editor,
+      expandToFullLine(editor, contentRange)
+    ).map(({ range }) => range);
+
+    const filteredTokenRanges = hasExplicitRange
+      ? this.filterRangesByIterationScope(contentRange, tokenRanges)
+      : tokenRanges;
+
+    return filteredTokenRanges.map(
+      (contentRange) =>
+        new TokenTarget({
+          editor,
+          isReversed,
+          contentRange,
+        })
```

Comment:
I'd be tempted to have this construction be a separate function in the scope handler.  Then we can avoid having to construct targets that will end up getting filtered out when the above filter step is in the base class.  Idk; I don't feel super strongly on this one tho. Might be simpler to just have one function that returns targets rather than one that returns the ranges and another that constructs the targets

One thought though is that returning something other than a target would allow us to add domains to each range, so that the base class can use that for intersection

Reply:
I've had that thought as well. The problem is that the constructor for some of these targets has a lot more information than this. Basically this is the untyped clone parameter issue all over again.

Reply:
Another advantage of separating the construction is that we can remove another arg from this function signature; don't need `isReversed` here

Would be kinda cool if this function just took an editor and a range

Reply:
Agreed. If we can just find a good enough solution for the constructor parameters then I'm totally with you.